third_party/BCVTB/utilXml.c

```diff
@@ -860,51 +857,72 @@ int check_variable_cfg_Validate(char const* const fileName) {
   char const* const jarPath = "/lib/xml/build/jar/DTDValidator.jar";
 #endif
 
-  BCVTB_HOME = getenv("BCVTB_HOME");
-  if (NULL == BCVTB_HOME) {
+  char const* const BCVTB_HOME = getenv("BCVTB_HOME");
+  if (BCVTB_HOME == NULL) {
     fprintf(stderr, "Error: Cannot get environment variable: BCVTB_HOME.\n");
     return -1;
   }
-  command = (char*)malloc(sizeof(char) * (strlen(BCVTB_HOME) + strlen(fileName) + strlen("variables.dtd") + 20) * 3);
-  if (NULL == command) {
-    fprintf(stderr,
-            "Error: Memory allocation failed in"
-            "       check_variable_cfg_Validate"
-            "       when parsing file '%s'.\n"
-            "       Program aborting.\n",
-            fileName);
-    return -1;
-  }
-  dtdFileName = (char*)malloc(sizeof(char) * (strlen(BCVTB_HOME) + 30));
-  if (NULL == command) {
```

Comment:
Note that this check on line 879 was wrong, it was checking the wrong variable...

Doesn't matter that much because malloc will never fail on the systems we run this on, but still